src/Tests/Microsoft.NET.TestFramework/OutputPathCalculator.cs

```diff
@@ -0,0 +1,252 @@
+
+                var ns = project.Root.Name.Namespace;
+
+                var propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+                var targetFrameworksElement = propertyGroup.Element(ns + "TargetFrameworks");
+                if (targetFrameworksElement != null)
+                {
+                    string targetFrameworks = targetFrameworksElement.Value;
+                    if (!targetFrameworks.Contains(';'))
+                    {
+                        targetFramework = targetFrameworks;
+                    }
+                }
+                else
+                {
+                    var targetFrameworkElement = propertyGroup.Element(ns + "TargetFramework");
+                    targetFramework = targetFrameworkElement?.Value;
+                }
+            }
+            return targetFramework ?? "";
+        }
+
+        public bool IsMultiTargeted()
+        {
+            if (!string.IsNullOrEmpty(TargetFramework) && TargetFramework.Contains(';'))
+            { 
+                return true;
+            }
+            var project = XDocument.Load(ProjectPath);
+
+            var ns = project.Root.Name.Namespace;
+
+            var propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+            var targetFrameworksElement = propertyGroup.Element(ns + "TargetFrameworks");
+            if (targetFrameworksElement != null)
+            {
+                string targetFrameworks = targetFrameworksElement.Value;
+                return !string.IsNullOrEmpty(targetFrameworks);
+            }
+
+            return false;
+
+        }
+
+        private bool UsesStandardOutputPaths()
+        {
+            if (UseStandardOutputPaths.HasValue)
+            {
+                return UseStandardOutputPaths.Value;
+            }
+
+            
+
+            //  If we end up enabling standard output paths when targeting a certain version of .NET or higher, the logic for the rest of this method would look something like this:s
+            //if (!IsSdkProject)
+            //{
+            //    return false;
+            //}
+
+            //string targetFramework = TryGetTargetFramework();
+
+            //if (targetFramework == null)
+            //{
+            //    return false;
+            //}
+
+            //if (IsMultiTargeted())
+            //{
+            //    return false;
+            //}
+
+            //var framework = NuGetFramework.Parse(targetFramework);
+            //if (framework.Framework != ".NETCoreApp")
+            //{
+            //    return false;
+            //}
+            //if (framework.Version.Major >= 8)
+            //{
+            //    return true;
+            //}
+            //else
+            //{
+            //    return false;
+            //}
+
+            return false;
+        }
+
+        public string GetOutputDirectory(string targetFramework = null, string configuration = "Debug", string runtimeIdentifier = "")
```

Comment:
When using artifacts output format, the logic is very similar and can be merged.  But otherwise the logic is different, so I thought it was simplest to leave them as separate functions.